### PR TITLE
Improvements for trailMesh

### PR DESCRIPTION
**Pull request by lockphase**
Added options to constructor, deprecated old constructor. Follows existing mesh creation practices for consistency and future use. 

Users can now control the trailMesh's segments, sections and switch off trail tapering when not needed. Defaults apply to preserve back-compat.

Updated trail mesh creation and update function to account for trail segments and tapering. Updated clone and parse functions to use new contructor.

Pls see forum [thread](https://forum.babylonjs.com/t/trail-mesh-with-uvs/47965) here, thanks!

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15125/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)
Important - these might fail sporadically. This is an optional test.

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15125/merge/testResults/webgpuplaywright/index.html

**Comment by Popov72** [comment]
You can keep backward compatibility with the old constructor by adding some decalarations. See how it's done for `VertexBuffer`:

https://github.com/BabylonJS/Babylon.js/blob/master/packages/dev/core/src/Buffers/buffer.ts#L470-L529

(you will have to add `autoStart` in the `option` object)

**Review by RaananW — DISMISSED** [review]
The trailmesh demo from the docs is not working as expected - https://playground.babylonjs.com/?snapshot=refs/pull/15125/merge#1F4UET#33

Would also be great to add a visualization test, if possible. You can add a playground to the config.json file and give a few frames to render the trail as well. If that's a problem we can do that later.

File: packages/dev/core/src/Meshes/trailMesh.ts [hunk 1]
```diff
@@ -6,12 +6,43 @@ import type { Scene } from "../scene";
...
+     */
+    sections?: number;
+    /**
+     * tapers the trailing mesh (default: false)
+     */
+    doNotTaper?: boolean;
+    /**
+     * automatically start trailing mesh. (default: true)
+     */
+    autoStart: boolean;
```
[the remark this answers is not in the document]
Comment (lockphase) [reply packages/dev/core/src/Meshes/trailMesh.ts:43]: Done!

File: packages/dev/core/src/Meshes/trailMesh.ts [hunk 2]
```diff
@@ -24,29 +55,54 @@ export class TrailMesh extends Mesh {
...
+            this.diameter = diameterOrOptions.diameter || 1;
+            this._length = diameterOrOptions.length || 60;
+            this._segments = diameterOrOptions.segments ? (diameterOrOptions.segments > this._length ? this._length : diameterOrOptions.segments) : this._length;
+            this._sectionPolygonPointsCount = diameterOrOptions.sections || 4;
+            this._doNotTaper = diameterOrOptions.doNotTaper || false;
+            this._autoStart = diameterOrOptions.autoStart || true;
+        } else {
+            this.diameter = diameterOrOptions || 1;
+            this._length = length;
+            this._autoStart = autoStart;
```
[the remark this answers is not in the document]
Comment (lockphase) [reply packages/dev/core/src/Meshes/trailMesh.ts:103]: Thanks, ```this._doNotTaper``` was also missing, fixed!

**Comment by lockphase** [comment]
> The trailmesh demo from the docs is not working as expected
I have tested the demo from the docs on local after the fix, it is working now. I presume another vis test is no longer needed? Also, there are 3 identical trails in the demo, perhaps setting different trail params per orb will be sufficient?

**Review by Popov72 — APPROVED** [review]

**Pull request merged**